source/state_representation/src/geometry/Ellipsoid.cpp

```diff
@@ -189,15 +189,15 @@ void Ellipsoid::set_data(const std::vector<double>& data) {
 }
 
 std::ostream& operator<<(std::ostream& os, const Ellipsoid& ellipsoid) {
+  auto prefix = ellipsoid.is_empty() ? "Empty ": "";
```

Comment:
While we are here, this checking behaviour is in fact copied in all derived classes. Could you make an helper function that would do that checking in `State` and return the first line? You will still have to check for emptiness and return but at least you don't have to copy this relatively complex behaviour in each classes.

Reply:
Hmm, not sure. This would have to be a public member of state or a public static function in state representation namespace, are we sure we want to clutter this with a helper like this? I'd prefer not to